### ISSUE TITLE
Split DNS through Subnet Router breaks with nftables

### ISSUE THREAD

**Issue by davidemyers** [tailscale/tailscale#8933]
### What is the issue?

All systems described below run Ubuntu Server 22.04 (so they use `iptables-nft`). These systems also all use `ufw`.

I use Split DNS to forward queries for my local domain to my home LAN router (which is not on my Tailnet). I have a Subnet Router on my LAN which advertises a host route (/32) to the router. I use the default ACLs.

Upgrading the Subnet Router to 1.48.0 causes DNS queries from remote clients for hostnames in my domain to time out:
```
dem@stratus:~$ host router
;; communications error to 127.0.0.53#53: timed out
;; communications error to 127.0.0.53#53: timed out
;; no servers could be reached
```
The result is the same querying 100.100.100.100 directly, as well as from clients running either 1.46.1 or 1.48.0. Remote clients can ping the local IP address of the LAN router so subnet routing still works.

Setting `TS_DEBUG_FIREWALL_MODE=iptables` on only the Subnet Router restores functionality.

Thanks.

### OS

Linux

### OS version

Ubuntu Server 22.04

### Tailscale version

1.48.0

### Other software

UFW

### Bug report

Will provide if relevant.

**Comment by catzkorn** [comment]
Hi @davidemyers

Could you email us at support@tailscale.com? We would like to look at this more closely and will need to know what the tailnet is.

Charlotte

**Comment by joaoportela** [comment]
Seems related to the changes they did to the routing in this release. 
Probably the same as here:
https://github.com/tailscale/tailscale/issues/8942

You can try downloading or the workaround that I suggested.

**Comment by raggi** [comment]
1.48.1 has been released and will now default to iptables again.

You can test out the newer "detection" which will pick between nftables and iptables based on usage, by setting: `TS_DEBUG_FIREWALL_MODE=auto` for example in `/etc/default/tailscaled`.

There is additionally a fix in 1.48.1 that resolves the conflict with ufw that lead to forwarding traffic (subnet routers and exit nodes) from functioning properly.

We plan to re-enable auto-detection in a future release.
